tools/test-ecosystem.mjs

```diff
@@ -0,0 +1,154 @@
+/**
+ * @fileoverview A utility to test an ecosystem plugin against the built ESLint.
+ * @author Josh Goldberg
+ */
+
+"use strict";
+
+import chalk from "chalk";
+import { $ } from "execa";
+import fs from "node:fs/promises";
+import path from "node:path";
+import util from "node:util";
+
+const plugins = new Map([
+	[
+		"eslint-plugin-unicorn",
+		{ repository: "https://github.com/sindresorhus/eslint-plugin-unicorn" },
+	],
+	["@eslint/css", { repository: "https://github.com/eslint/css" }],
+	["@eslint/json", { repository: "https://github.com/eslint/json" }],
+	["@eslint/markdown", { repository: "https://github.com/eslint/markdown" }],
+	[
+		"@eslint-community/eslint-plugin-eslint-comments",
+		{
+			repository:
+				"https://github.com/eslint-community/eslint-plugin-eslint-comments",
+		},
+	],
+	[
+		"eslint-plugin-vue",
+		{ repository: "https://github.com/vuejs/eslint-plugin-vue" },
+	],
+	[
+		"typescript-eslint",
+		{
+			linker: `na link ${process.cwd()}`,
+			repository:
+				"https://github.com/typescript-eslint/typescript-eslint",
+		},
+	],
+]);
```

Comment:
I suggest pin commit hash, and update with an automated job. Repos can be broken by accidental commits.

Reply:
Good call! I moved the data out to a `plugins.json` and added a `"commit"` for each plugin.

Suggestion: do we want to split auto-updating of those commit hashes to a followup PR? I don't know of an existing pre-made action that does exactly what we'd want here for creating a bump PR for arbitrary JSON if one doesn't already exist.

Reply:
https://github.com/peter-evans/create-pull-request

Reply:
Heh, good point. For the rest of it I added a `test:ecosystem:update` script and `ecosystem-updates.yml`.